tests/v3/compatibility_suite/test_v3_message_consumer.py

```diff
@@ -0,0 +1,663 @@
+"""V3 Message consumer feature tests."""
+
+from __future__ import annotations
+
+import ast
+import json
+import logging
+import re
+from typing import TYPE_CHECKING, Any, List, NamedTuple
+
+from pytest_bdd import (
+    given,
+    parsers,
+    scenario,
+    then,
+    when,
+)
+
+from tests.v3.compatibility_suite.util import (
+    FIXTURES_ROOT,
+    PactInteractionTuple,
+    parse_markdown_table,
+)
+from tests.v3.compatibility_suite.util.consumer import (
+    a_message_integration_is_being_defined_for_a_consumer_test,
+)
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from pact.v3.pact import AsyncMessageInteraction, InteractionVerificationError
+
+logger = logging.getLogger(__name__)
+
+################################################################################
+## Helpers
+################################################################################
+
+
+class ReceivedMessage(NamedTuple):
+    """Holder class for Message Received Payload."""
+
+    body: Any
+    context: Any
+
+
+class PactResult(NamedTuple):
+    """Holder class for Pact Result objects."""
+
+    messages: List[ReceivedMessage]
+    pact_data: dict[str, Any] | None
+    errors: List[InteractionVerificationError]
+
+
+def assert_type(expected_type: str, value: Any) -> None:  # noqa: ANN401
+    logger.debug("Ensuring that %s is of type %s", value, expected_type)
+    if expected_type == "integer":
+        assert value is not None
+        assert isinstance(value, int) or re.match(r"^\d+$", value)
+    else:
+        msg = f"Unknown type: {expected_type}"
+        raise ValueError(msg)
```

Comment:
since this only asserts whether or not the value is an `integer`, should we rename it to `assert_is_integer` or something like that?

Reply:
I did consider that, but I figured there might be a need to check for other types in other parts of the compatibility suite; in which case I would like to move the declaration of this function out into the `util` module 🙂

But I won't do that prematurely...

Reply:
How about we change it now and, should there be a need to check for other types later, it can be refactored?